yarn-project/noir-libs/noir-aztec/src/types/vec.nr

```diff
@@ -37,18 +37,46 @@ impl<T, MaxLen> BoundedVec<T, MaxLen> {
         self.len -= 1;
         elem
     }
+
+    fn any<Env>(self, predicate: fn[Env](T) -> bool) -> bool {
+        let mut ret = false;
+        for i in 0..MaxLen {
+            if (i < self.len as u64) {
+                ret |= predicate(self.storage[i]);
+            }
+        }
```

Comment:
```suggestion
        let exceeded_len = false;
        for i in 0..MaxLen {
            exceeded_len |= (i == self.len); // check if the loop has gone beyond the len of the vec 
            if (!exceeded_len) {
                ret |= predicate(self.storage[i]);
            }
        }
```
This might be more efficient, because it avoids having to do range checks with `<`. I might have got my boolean logic slightly wrong, because I rushed this suggestion, though :)

Reply:
Oooh right that's brilliant! Will change :)